app/code/Magento/Cms/Model/PageRepository.php

```diff
@@ -152,6 +152,11 @@ public function save(\Magento\Cms\Api\Data\PageInterface $page)
         }
         try {
             $this->validateLayoutUpdate($page);
+            if($page->getId()){
+                $savedPage = $this->getById($page->getId());
+                $pageData = array_replace_recursive($savedPage->toArray(),$page->toArray());
+                $page->addData($pageData);
```

Comment:
Magento\Cms\Api\Data\PageInterface does not have toArray/addData methods. Can you please use hydrator to extract and add data